File: RestaurantFinder/src/main/java/com/example/restaurant_finder/service/CustomerService.java
package com.example.restaurant_finder.service;

import org.springframework.http.ResponseEntity;

import com.example.restaurant_finder.dto.CustomerLogin;

public interface CustomerService {
	
	public ResponseEntity<CustomerLogin> getCustomerByCusomerEmail(CustomerLogin login);
	
	public ResponseEntity<String> logoutCustomer(CustomerLogin login);
	
	public ResponseEntity<CustomerLogin> addCustomer(CustomerLogin register);
	
	public CustomerLogin updateProfile(CustomerLogin customer);
	
	public CustomerLogin getPassword(CustomerLogin customer);
	
	public ResponseEntity<String> updatePassword(CustomerLogin customer);
}

File: src/app/services/customer.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import { Customer } from '../customer-login/customer.interface';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  constructor(private http: HttpClient) { }

  loginCustomer(customer: Customer) {
    return this.http.post<Customer>("http://localhost:8080/customerLogin",customer);
  }

  logoutCustomer(customer: Customer) {
    return this.http.post<Customer>("http://localhost:8080/customerLogout", customer);
  }

  addCustomer(customer: Customer) {
    return this.http.post<Customer>("http://localhost:8080/customerRegistration", customer);
  }

  updateProfile(customer: Customer) {
    return this.http.post<Customer>("http://localhost:8080/updateProfile", customer);
  }

  getPassword(customer: Customer) {
    return this.http.post<Customer>("http://localhost:8080/getPassword", customer);
  }

  updatePassword(customer: Customer) {
    return this.http.post<Customer>("http://localhost:8080/updatePassword", customer);
  }
}
